### PR TITLE
Add clock skew validation

### PR DESCRIPTION
**Pull request by tsorya**
Till "known" timestamp will be sent by inventory check and while we will move to known we will have timestampCheck step.

### CODE REVIEW

File: internal/host/host.go [hunk 1]
```diff
@@ -183,6 +185,28 @@ func (m *Manager) UpdateInventory(ctx context.Context, h *models.Host, inventory
 	return m.db.Model(h).Update("inventory", inventory).Error
 }
 
+func (m *Manager) UpdateTimestamp(ctx context.Context, h *models.Host, timestamp int64) error {
```
Comment (avishayt) [inline internal/host/host.go:188]: When we spoke about having the timestamp as part of inventory, I assumed the agent would be sending it together with the inventory.  If you're sending it separately, which is fine, I would put timestamp as a separate property.  What do you think?

Comment (ori-amizur) [reply internal/host/host.go:188]: Agree.

File: internal/host/validator.go [hunk 2]
```diff
@@ -395,6 +396,42 @@ func (v *validator) printHostnameUnique(c *validationContext, status validationS
...
+	}
+	for _, h := range c.cluster.Hosts {
+		if h.ID.String() != c.host.ID.String() && h.Inventory != "" && *h.Status != models.HostStatusDisconnected {
+			var otherInventory models.Inventory
+			if err := json.Unmarshal([]byte(h.Inventory), &otherInventory); err != nil {
+				v.log.WithError(err).Warnf("Illegal inventory for host %s", h.ID.String())
+				continue
+			}
+
+			if math.Abs(float64(c.inventory.Timestamp-otherInventory.Timestamp))/60 > float64(v.hwValidatorCfg.MaximumAllowedTimeDiffMinutes) {
```
Comment (ori-amizur) [inline internal/host/validator.go:414]: If there is one host with time skew, all the hosts will fail this validation.  Is that what we want?

File: internal/host/host.go (hunk 1 above)
Comment (tsorya) [reply internal/host/host.go:188]: we will send it as part of inventory. 
i just need an option to send it without inventory too. 
As i wrote in description we will get it in 2 ways cause we are not running inventory check on Known state but we need to get timestamp.

File: internal/host/validator.go (hunk 2 above)
Comment (tsorya) [reply internal/host/validator.go:414]: it is what was agreed for now

File: swagger.yaml [hunk 3]
```diff
@@ -2744,6 +2745,9 @@ definitions:
         $ref: '#/definitions/memory'
       cpu:
         $ref: '#/definitions/cpu'
+      timestamp:
+        type: integer
```
Comment (ori-amizur) [inline swagger.yaml:2749]: Why is the timestamp integer?

File: internal/host/host.go (hunk 1 above)
Comment (tsorya) [reply internal/host/host.go:188]: we can set as new host property. though it will be less readable to have it as some  new strange host field

Comment (tsorya) [reply internal/host/host.go:188]: moving all to inventory

**Comment by tsorya** [comment]
will reopen

**Pull request closed without merging**